### ISSUE TITLE
`client` parameter in `Consumer` constructor doesn't work as documented 

### ISSUE THREAD

**Issue by prkumar** [prkumar/uplink#26]
## Precondition

Consider the following consumer:
```python
class GitHub(uplink.Consumer):

    @uplink.get("/users/{username}")
    def get_user(self, username):
        """Get a single user."""
```

## Steps to recreate

Instantiate this consumer with a specific client instance:
```python
GitHub(base_url="https://api.github.com/", client=uplink.RequestsClient())
```

## Expected

Consumer instance builds properly and uses the given client instance.

**Note: when the `client` parameter is given a `uplink.clients.interfaces.HttpClientAdapter` subclass, it should instantiate a client instance; otherwise the provided value should be used as given.**

## Actual

Exception raised on instantiation:
```python
Traceback (most recent call last):
  File "/Users/prkumar/Library/Preferences/PyCharm2017.2/scratches/scratch_1.py", line 11, in <module>
    GitHub(base_url="https://api.github.com/", client=uplink.RequestsClient())
  File "/Users/prkumar/Developer/uplink/uplink/builder.py", line 170, in __init__
    self._build(builder)
  File "/Users/prkumar/Developer/uplink/uplink/builder.py", line 175, in _build
    caller = call_builder.build(self, definition_builder)
  File "/Users/prkumar/Developer/uplink/uplink/builder.py", line 149, in build
    RequestPreparer(self, definition),
  File "/Users/prkumar/Developer/uplink/uplink/builder.py", line 42, in __init__
    if issubclass(self._client, clients.interfaces.HttpClientAdapter):
TypeError: issubclass() arg 1 must be a class
```

**Comment by kadrach** [comment]
Not sure if I'm understanding the intent here properly, but simply removing the failing checks works:

```diff
diff --git a/uplink/builder.py b/uplink/builder.py
index d9dd21b..1335ff3 100644
--- a/uplink/builder.py
+++ b/uplink/builder.py
@@ -39,8 +39,6 @@ class RequestPreparer(object):
     def __init__(self, uplink_builder, definition):
         self._hook = uplink_builder.hook
         self._client = uplink_builder.client
-        if issubclass(self._client, clients.interfaces.HttpClientAdapter):
-            self._client = self._client()
         self._base_url = str(uplink_builder.base_url)
         self._converter_registry = self._make_converter_registry(
             uplink_builder, definition
```
Given you pass in an instance in your example, `issubclass` fails because it expects `self._client.__class__`. 

Apart from this, uplink works beautifully for my usecase of passing in a predefined requests-oauthlib session:

```python
session = OAuth2Session(...)  # Gets me token refresh for free

api = SomeAPI(base_url=BASE_URL, client=uplink.RequestsClient(session=session))
api.do_the_thing(...)
```

**Comment by prkumar** [comment]
@kadrach, looks good!

The original intent of implementing this check was to silently support an uninitialized subclass of `uplink.clients.interfaces.HttpClientAdapter` in lieu of a client instance. For example:
```python
# This...
api = SomeAPI(base_url=BASE_URL, client=uplink.RequestsClient)

# Should be equivalent to...
api = SomeAPI(base_url=BASE_URL, client=uplink.RequestsClient())
```
But, I'm fine with forgoing support for the former case, in which we initialize the instance for the user when given a subclass. 

Do you want to open a PR to fix this? We'll need to release a patch version with these changes, so if you have the time to contribute, please cherry-pick your changes onto `stable`.

**Comment by prkumar** [comment]
Closed by #29
